src/IO/configuration_XML.cpp

```diff
@@ -283,6 +319,75 @@ void writeConfiguration(const std::string &filename, const std::string &rootname
     xmlFreeTextWriter(writer);
 }
 
+/*!
+    Write the Config Tree to a c++ string (xml-stringfication). All contents will 
+    be appended to the target source string.
+    The method is meant for general-purpose xml info flushing.  
+    \param source string to write to
+    \param rootConfig pointer to the Config tree to be stringfied.
+    \param rootname (optional) name of the root section. Default is "root".
+*/
+void writeBufferConfiguration(std::string &source, const Config *rootConfig, const std::string &rootname)
```

Comment:
Would it be possible to add a private function that takes in input a xmlTextWriterPtr? This will allow to share code between writeConfiguration and writeBufferConfiguration? For "symmetry", this apply also to "readBufferConfiguration".

In bitpit , arguments that are modified by the function are usually passed by pointer at the end of the argument list.

Do we really need a function with a different name? The signature of writeConfiguration and writeBufferConfiguration is different, we can use the same name.

I would still pass a "format" argument (also in the read* function). In this way the function that write the configuration to file and the one that write the configuration to string will generate the same output.

Reply:
You're right a compact code is always better. But since pending pull https://github.com/optimad/bitpit/pull/267 and needs of issue #269 suggest a major underhood rework of tree interface and its parsing capabilities, and the introduction of rapidXML will make a lot easier to compact the code w.r.t. to libxml2, I think it's not the case to lose much time on this now. Anyway, as you prefer, if you want to fix it right now, i'll fix it. Let me know